Student/Student/Profile/ProfilePresenter.swift

```diff
@@ -0,0 +1,155 @@
+                guard settings == nil else { return }
+                NotificationCenter.default.post(name: NSNotification.Name("redux-action"), object: nil, userInfo: [
+                    "type": "userInfo.updateUserSettings",
+                    "error": true,
+                    "payload": [ "hideOverlay": !colorOverlay ],
+                ])
+            }
+        })
+
+        if let root = helpLinks.first, helpLinks.count > 1 {
+            cells.append(ProfileViewCell("help", name: root.text) { [weak self] cell in
+                self?.view?.showHelpMenu(from: cell)
+            })
+        }
+        cells.append(ProfileViewCell("settings", name: NSLocalizedString("Settings", comment: "")) { [weak self] _ in
+            self?.view?.route(to: .profileSettings, options: [.modal, .embedInNav])
+        })
+        if showDevMenu {
+            cells.append(ProfileViewCell("developerMenu", name: NSLocalizedString("Developer Menu", comment: "")) { [weak self] _ in
+                self?.view?.route(to: .developerMenu, options: [.modal, .embedInNav])
+            })
+        }
+        if canActAsUser {
+            cells.append(ProfileViewCell("actAsUser", name: NSLocalizedString("Act as User", comment: "")) { [weak self] _ in
+                self?.view?.route(to: .actAsUser, options: [.modal, .embedInNav])
+            })
+        }
+        cells.append(ProfileViewCell("changeUser", name: NSLocalizedString("Change User", comment: "")) { _ in
+            guard let delegate = UIApplication.shared.delegate as? LoginDelegate else { return }
+            delegate.changeUser()
+        })
+        if env.currentSession?.actAsUserID != nil {
+            cells.append(ProfileViewCell("logOut", name: NSLocalizedString("Stop Act as User", comment: "")) { _ in
+                guard let delegate = UIApplication.shared.delegate as? AppDelegate else { return }
+                guard let session = delegate.environment.currentSession else { return }
+                delegate.stopActing(as: session)
+            })
+        } else {
+            cells.append(ProfileViewCell("logOut", name: NSLocalizedString("Log Out", comment: "")) { _ in
+                guard let delegate = UIApplication.shared.delegate as? AppDelegate else { return }
+                guard let session = delegate.environment.currentSession else { return }
+                delegate.userDidLogout(session: session)
+            })
+        }
+        return cells
+    }
+
+    init(env: AppEnvironment = .shared) {
+        self.env = env
+    }
+
+    public func viewIsReady() {
+        helpLinks.refresh(force: true)
+        permissions.refresh(force: true)
+        settings.refresh(force: true)
+        tools.refresh(force: true)
+    }
+
+    public func didTapVersion() {
+        showDevMenu = true
+        UserDefaults.standard.set(true, forKey: "showDevMenu")
+        self.view?.reload()
+    }
+}
```

Comment:
Each of these ProfilePresenters are very very similar. It seems a few `if app ==` would be easier to maintain, rather than triplicating most of the logic. What do you think?